### PR TITLE
Basic DAE Initialization

### PR DESCRIPTION
**Pull request by kanav99**
https://cse.cs.ucsb.edu/sites/default/files/publications/SCE001495.pdf

Using NLsolve right now, I will try to remove that out

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -382,6 +383,10 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.
                              isout,reeval_fsal,
                              u_modified,opts,destats)
   if initialize_integrator
+    if isdae
+      initialize_dae!(integrator, u, du, prob.differential_vars, initializealg, Val(isinplace(prob)))
+      @show du
```
Comment (ChrisRackauckas) [inline src/solve.jl]:
```suggestion
```

File: src/initialize_dae.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		else
+			u[i] = r.zero[i]
+		end
+	end
+end
+
+function initialize_dae!(integrator, u, du, differential_vars, alg::BrownFullBasicInit, ::Val{true})
+	@unpack p, t, f = integrator
+
+	nlequation = (x) -> begin
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl]:
```suggestion
	nlequation = (dx,x) -> begin
```

File: src/initialize_dae.jl [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	for i in 1:length(u)
+		if differential_vars[i]
+			du[i] = r.zero[i]
+		else
+			u[i] = r.zero[i]
+		end
+	end
+end
+
+function initialize_dae!(integrator, u, du, differential_vars, alg::BrownFullBasicInit, ::Val{true})
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl]:
```suggestion
function initialize_dae!(integrator, u, du, differential_vars, alg::BrownFullBasicInit, ::Val{false})
```

File: src/initialize_dae.jl [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+	nlequation = (x) -> begin
+		for i in 1:length(x)
+			if differential_vars[i]
+				du[i] = x[i]
+			else
+				u[i] = x[i]
+			end
+		end
+		f(du, u, p, t)
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl]:
```suggestion
		du .= f(u, p, t)
```

**Comment by ChrisRackauckas** [comment]
Needs a test.

**Comment by kanav99** [comment]
Added some basic test. There is a failing test (non index 1 DAE), that means we know whats missing. I guess we need an interface to define what all is known to the user, and what all the user wants us to initialize

**Comment by ChrisRackauckas** [comment]
>I guess we need an interface to define what all is known to the user, and what all the user wants us to initialize

That can come from the initialization method choice?

File: src/solve.jl [hunk 5]
```diff
@@ -382,6 +383,9 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.
                              isout,reeval_fsal,
                              u_modified,opts,destats)
   if initialize_integrator
+    if isdae
```
Comment (ChrisRackauckas) [inline src/solve.jl]: evaluate to see if initialization is required first? Or do that in initialize?

**Pull request merged**